packages/jsii/lib/assembler.ts

```diff
@@ -628,6 +630,29 @@ export class Assembler implements Emitter {
 
         this._visitDocumentation(symbol, method);
 
+        // If the last parameter is a datatype, verify that it does not share any field names with
+        // other function arguments, so that it can be turned into keyword arguments by jsii frontends
+        // that support such.
+        const lastParamTypeRef = apply(last(parameters), x => x.type);
+        const lastParamSymbol = last(signature.getParameters());
+        if (lastParamTypeRef && spec.isNamedTypeReference(lastParamTypeRef)) {
+            this._deferUntilTypesAvailable(symbol.name, [lastParamTypeRef], lastParamSymbol!.declarations[0], (lastParamType) => {
```

Comment:
I am worried about this programming model. I really liked your idea of phases as a safer and more deterministic alternative.